traffic.sh

```diff
@@ -3,41 +3,180 @@
 # Licensed under the BSD 3-Clause License.
 # SPDX-License-Identifier: BSD-3-Clause
 
+# look for clab files
+shopt -s nullglob
+clabfiles=(*.clab.yml)
 
-set -eu
+
+# functions to deal with yml files, 
+# shoutout to Stefan Farestam
+# https://stackoverflow.com/questions/5014632/how-can-i-parse-a-yaml-file-from-a-linux-shell-script
+
+parse_yaml() {
+    local yaml_file=$1
+    local prefix=$2
+    local s
+    local w
+    local fs
+
+    s='[[:space:]]*'
+    w='[a-zA-Z0-9_.-]*'
+    fs="$(echo @ | tr @ '\034')"
+
+    (
+        sed -e '/- [^\“]'"[^\']"'.*: /s|\([ ]*\)- \([[:space:]]*\)|\1-\'$'\n''  \1\2|g' |
+            sed -ne '/^--/s|--||g; s|\"|\\\"|g; s/[[:space:]]*$//g;' \
+                -e 's/\$/\\\$/g' \
+                -e "/#.*[\"\']/!s| #.*||g; /^#/s|#.*||g;" \
+                -e "s|^\($s\)\($w\)$s:$s\"\(.*\)\"$s\$|\1$fs\2$fs\3|p" \
+                -e "s|^\($s\)\($w\)${s}[:-]$s\(.*\)$s\$|\1$fs\2$fs\3|p" |
+            awk -F"$fs" '{
+            indent = length($1)/2;
+            if (length($2) == 0) { conj[indent]="+";} else {conj[indent]="";}
+            vname[indent] = $2;
+            for (i in vname) {if (i > indent) {delete vname[i]}}
+                if (length($3) > 0) {
+                    vn=""; for (i=0; i<indent; i++) {vn=(vn)(vname[i])("_")}
+                    printf("%s%s%s%s=(\"%s\")\n", "'"$prefix"'",vn, $2, conj[indent-1], $3);
+                }
+            }' |
+            sed -e 's/_=/+=/g' |
+            awk 'BEGIN {
+                FS="=";
+                OFS="="
+            }
+            /(-|\.).*=/ {
+                gsub("-|\\.", "_", $1)
+            }
+            { print }'
+    ) <"$yaml_file"
+}
```

Comment:
This is cool, buuuut I wonder how many people out there understand whats happening here.
I'd rather suggest you look at [yq](https://github.com/mikefarah/yq) to easily parse yaml and leverage its container image to do this without any dependencies.

Reply:
I was specifically trying to avoid more containers/dependencies, this is a pure (excluding standard grep/awk/sed/tr utils) bash script implementation to be able to parse yml into bash variables.